src/test/unit/EigenPodManagerUnit.t.sol

```diff
@@ -117,220 +85,438 @@ contract EigenPodManagerUnitTests is Test, EigenPodPausingConstants {
             )
         );
 
+        // Set beaconChainETHStrategy
         beaconChainETHStrategy = eigenPodManager.beaconChainETHStrategy();
 
-        // excude the zero address, the proxyAdmin and the eigenPodManager itself from fuzzed inputs
+        // Set defaultPod
+        defaultPod = eigenPodManager.getPod(defaultStaker);
+
+        // Exclude the zero address, the proxyAdmin and the eigenPodManager itself from fuzzed inputs
         addressIsExcludedFromFuzzedInputs[address(0)] = true;
         addressIsExcludedFromFuzzedInputs[address(proxyAdmin)] = true;
         addressIsExcludedFromFuzzedInputs[address(eigenPodManager)] = true;
     }
 
-    function testRecordBeaconChainETHBalanceUpdateFailsWhenNotCalledByEigenPod(address improperCaller) public filterFuzzedAddressInputs(improperCaller) {
-        address staker = address(this);
-        IEigenPod eigenPod = _deployEigenPodForStaker(staker);
-        cheats.assume(improperCaller != address(eigenPod));
+    /*******************************************************************************
+                            Helper Functions/Modifiers
+    *******************************************************************************/
 
-        cheats.expectRevert(bytes("EigenPodManager.onlyEigenPod: not a pod"));
-        cheats.startPrank(address(improperCaller));
-        eigenPodManager.recordBeaconChainETHBalanceUpdate(staker, int256(0));
-        cheats.stopPrank();
+    function _initializePodWithShares(address podOwner, int256 shares) internal {
+        _deployAndReturnEigenPodForStaker(podOwner);
+        // Signature of `podOwnerShares(address)
+        bytes4 signature = 0x60f4062b;
+        bytes32 sharesToSet = bytes32(uint256(shares));
+        stdstore.target(address(eigenPodManager)).sig(signature).with_key(podOwner).checked_write(sharesToSet);
+    }
```

Comment:
is there a reason for using stdstore here instead of calling the `recordBeaconChainETHBalanceUpdate`?

seems potentially convenient for some setup but also like it could result in situations in which we are simulating otherwise inaccessible states (e.g. if `shares` is a non-whole Gwei amount, and we are otherwise successfully enforcing that user shares are always a whole-Gwei amount)

not trying to bash this, it's cool, honestly just trying to understand + offer my perspective

Reply:
My philosophy here is to create a pre-test state without having to depend on other functions.

You do make a good point on enforcing that `shares` is a non-whole Gwen amount, which we should do in this function.

Reply:
Although now that I think about it, if we ever change `recordBeaconChainEThBalanceUpdate` logic, then the tests should be updated to reflect that. Will update this to not use std store - thanks for the suggestion.

Reply:
cool, makes sense + works for me